hw/drivers/sensors/dps368/src/dps368.c

```diff
@@ -0,0 +1,1048 @@
+	if (rc) {
+		return rc;
+	}
+
+	DPS368_LOG(INFO,"DPS368init:sensor_set_interface:OK\n");
+
+	rc = sensor_mgr_register(sensor);
+	if (rc) {
+		return rc;
+	}
+
+	DPS368_LOG(INFO,"DPS368init:sensor_mgr_register:OK\n");
+
+#if !MYNEWT_VAL(BUS_DRIVER_PRESENT)
+	if (sensor->s_itf.si_type == SENSOR_ITF_SPI) {
+		rc = hal_spi_config(sensor->s_itf.si_num, &spi_dps368_settings_s);
+		if (rc == EINVAL) {
+			return rc;
+		}
+
+		rc = hal_spi_enable(sensor->s_itf.si_num);
+		if (rc) {
+			return rc;
+		}
+
+		rc = hal_gpio_init_out(sensor->s_itf.si_cs_pin, 1);
+		if (rc) {
+			return rc;
+		}
+	}
+#endif
+
+	return 0;
+}
+
+
+int dps368_config(struct dps368 *dps368, struct dps368_cfg_s *cfg)
+{
+
+	int rc;
+	struct sensor_itf *itf;
+
+	uint8_t sensor_exist;
+	uint8_t ready;
+
+	struct dps368_cfg_s updated_cfg = dps368->cfg;
+
+	itf = SENSOR_GET_ITF(&(dps368->sensor));
+
+	if(cfg->config_opt & DPS3xx_CONF_WITH_INIT_SEQUENCE){
+		updated_cfg.odr_t = cfg->odr_t;
+		updated_cfg.odr_p = cfg->odr_p;
+		updated_cfg.osr_t = cfg->osr_t;
+		updated_cfg.osr_p = cfg->osr_p;
+		goto INIT_SEQUENCE;
+	}else if (cfg->config_opt & DPS3xx_RECONF_ALL){
+		updated_cfg.odr_t = cfg->odr_t;
+		updated_cfg.odr_p = cfg->odr_p;
+		updated_cfg.osr_t = cfg->osr_t;
+		updated_cfg.osr_p = cfg->osr_p;
+		goto RECONF;
+	}else if (cfg->config_opt & DPS3xx_RECONF_ODR_ONLY){
+		updated_cfg.odr_t = cfg->odr_t;
+		updated_cfg.odr_p = cfg->odr_p;
+	}else if (cfg->config_opt & DPS3xx_RECONF_ODR_WITH_BEST_OSR){
+		updated_cfg.odr_t = cfg->odr_t;
+		updated_cfg.odr_p = cfg->odr_p;
+		updated_cfg.osr_t = dps368_get_best_osr(cfg->odr_t);
+		updated_cfg.osr_p = dps368_get_best_osr(cfg->odr_p);
+		goto RECONF;
+	}else if (cfg->config_opt & DPS3xx_RECONF_OSR_ONLY){
+		updated_cfg.osr_t = cfg->osr_t;
+		updated_cfg.osr_p = cfg->osr_p;
+		goto RECONF;
+	}else if (cfg->config_opt & DPS3xx_RECONF_OSR_ONLY){
+		goto RECONF_MODE_ONLY;
+	}
+
+
+INIT_SEQUENCE:
+	if((rc=dps368_verify_sensor(itf,&sensor_exist))){
+		return rc;
+	}
+	if (sensor_exist == 1){
+		DPS368_LOG(INFO,"DPS368:Found during config init sequence\n");
+	}
+
+	if((rc=dps368_soft_reset(&dps368->sensor))){
+			return rc;
+	}
+
+	DPS368_LOG(INFO,"DPS368:Soft reset: OK\n");
+
+	if((rc=dps368_is_trim_complete(itf,&ready))){
+				return rc;
+	}
+
+	if (ready == 1){
+
+		if((rc=dps368_prepare_calib_coeff(itf,&dps368->calib_coeffs,&dps368->temp_src))){
+			return rc;
+		}
+
+		DPS368_LOG(INFO,"DPS368:Calibration data prepared\n");
+	}
+
+RECONF:
+
+	if((rc=dps368_reconfig(dps368,updated_cfg.osr_t,updated_cfg.osr_p,updated_cfg.odr_t,updated_cfg.odr_p))){
+			return rc;
+		}
+
+	DPS368_LOG(INFO,"DPS368:Reconfig done\n");
+
+	if(cfg->config_opt & DPS3xx_RECONF_ALL)
+		goto RECONF_MODE_ONLY;
+	else
+		return 0;
+
+RECONF_MODE_ONLY:
+
+	if((rc=dps368_set_mode(dps368,cfg->mode))){
+			return rc;
+	}
+	DPS368_LOG(INFO,"DPS368:Mode set\n");
+
+	if((rc=sensor_set_type_mask(&(dps368->sensor), cfg->chosen_type))){
+				return rc;
+	}
+
+
+return 0;
```

Comment:
spaces

Reply:
Coding style related issues are addressed in new commits and are already fixed now.